src/main/java/com/dani/blog/base/service/BaseService.java

```diff
@@ -0,0 +1,52 @@
+package com.dani.blog.base.service;
+
+import com.dani.blog.base.data.DAO;
+import javassist.NotFoundException;
+import org.springframework.dao.EmptyResultDataAccessException;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Base implementation for {@link Service}
+ * @param <E> Entity
+ * @param <D> DTO
+ * @param <I> Identifier
+ */
+public class BaseService<E, D, I extends Serializable> implements Service<E, D, I> {
+    protected final Transformer<E, D> transformer;
+    protected final DAO<E, I> repository;
```

Comment:
It is also a good approach that instead of defining the repository and expecting it to be provided in the constructor, you create an `abstract protected DAO<E, I> getRepository()` method that is being used everywhere instead of the property. Using that approach would make it possible to define the method as `protected CommentRepository getRepository()` in CommentService and not needing casting.